src/option.h

```diff
@@ -478,19 +478,56 @@ namespace librealsense
         std::function<void(const option&)> _recording_function = [](const option&) {};
     };
 
+    class proxy_option : public option
+    {
+    public:
+        const char* get_value_description(float val) const override
+        {
+            return _proxy->get_value_description(val);
+        }
+        const char* get_description() const override
+        {
+            return _proxy->get_description();
+        }
+        virtual void set(float value) {}
+
+        float query() const override
+        {
+            return _proxy->query();
+        }
+
+        option_range get_range() const override
+        {
+            return _proxy->get_range();
+        }
+
+        bool is_enabled() const override
+        {
+            return  _proxy->is_enabled();
+        }
+
+        bool is_read_only() const override
+        {
+            return  _proxy->is_read_only();
+        }
+
+        explicit proxy_option(std::shared_ptr<option> proxy_option)
+            : _proxy(proxy_option)
+        {}
+        void enable_recording(std::function<void(const option &)> record_action) override
+        {
+            _recording_function = record_action;
+        }
+    protected:
+        std::shared_ptr<option> _proxy;
+        std::function<void(const option&)> _recording_function = [](const option&) {};
```

Comment:
Remove the `;`

Reply:
why? removing it will cause compilation error

Reply:
Oh I thought it was a regular function, my mistake :)